File: HB/20200714/18258.cpp
// push X: 정수 X를 큐에 넣는 연산이다.
// pop: 큐에서 가장 앞에 있는 정수를 빼고, 그 수를 출력한다. 만약 큐에 들어있는 정수가 없는 경우에는 -1을 출력한다.
// size: 큐에 들어있는 정수의 개수를 출력한다.
// empty: 큐가 비어있으면 1, 아니면 0을 출력한다.
// front: 큐의 가장 앞에 있는 정수를 출력한다. 만약 큐에 들어있는 정수가 없는 경우에는 -1을 출력한다.
// back: 큐의 가장 뒤에 있는 정수를 출력한다. 만약 큐에 들어있는 정수가 없는 경우에는 -1을 출력한다.

#include <iostream>
#include <queue>
#include <string>
using namespace std;

int main(void) {
    cin.tie(0);
    cin.sync_with_stdio(0);
	int num, data;
	queue<int> q; 
    string str;
	cin >> num;

	for (int i = 0; i < num; i++) {
		cin >> str;

		if (str == "push") {
			cin >> data;
			q.push(data);
		}
		else if (str == "pop") {
			if (q.size() == 0) std::cout << "-1\n";
			else {
				data = q.front();
				q.pop();
				cout << data << "\n";
			}
		}
		else if (str == "size") {
			cout << q.size() << "\n";
		}
		else if (str == "empty") {
			cout << q.empty() << "\n";
		}
		else if (str == "front") {
			if (q.size() == 0) std::cout << "-1\n";
			else cout << q.front() << "\n";
		}
		else if (str == "back") {
			if (q.size() == 0) std::cout << "-1\n";
			else cout << q.back() << "\n";
		}
	}
	return 0;
}

File: JY/7/7.14/18258.py
import sys
from collections import deque
input = sys.stdin.readline
N = int(input())
li = []

def push(queue, n):
    queue.append(n)

def front(queue):
    if len(queue)==0:
        print(-1)
    else:
        print(queue[0])

def back(queue):
    if len(queue)==0:
        print(-1)
    else:
        print(queue[-1])
def pop(queue):
    if len(queue)==0:
        print(-1)
    else:
        print(queue.popleft())

def size(queue):
    print(len(queue))

def empty(queue):
    if len(queue)==0:
        print(1)
    else:
        print(0)


q= deque()
for _ in range(N):
    li = input().split()
    if li[0] == 'push':
        push(q, int(li[1]))
    if li[0] == 'front':
        front(q)
    if li[0] == 'pop':
        pop(q)
    if li[0] == 'empty':
        empty(q)
    if li[0] == 'back':
        back(q)
    if li[0] == 'size':
        size(q)
    li.clear()
